File: app/api/apps.py
from fastapi import APIRouter, HTTPException
from dependencies import app_manager
app_router = APIRouter()

@app_router.get("/active")
async def get_active_app():
    active_app = app_manager.get_current_app()
    if active_app:
        return {"active_app": active_app.name}
    else:
        return {"active_app": None}
    
@app_router.post("/activate/{app_name}")
async def activate_app(app_name: str):
    if app_manager.set_active_app_by_name(app_name):
        return {"status": "ok", "active_app": app_name}
    else:
        raise HTTPException(status_code=404, detail="App not found") 
    
@app_router.get("/available")
async def get_available_apps():
    apps = app_manager.get_available_apps()
    return {"available_apps": [app.name for app in apps]}

File: webui/src/lib/api/apps.ts
const API_URL = process.env.NEXT_PUBLIC_API_URL || "http://localhost:8000";

export interface AppResponse {
  active_app: string | null;
}

export interface AvailableAppsResponse {
  available_apps: string[];
}

export interface ActivateAppResponse {
  status: string;
  active_app: string;
}

export async function getActiveApp(): Promise<string | null> {
  const response = await fetch(`${API_URL}/apps/active`);
  if (!response.ok) {
    throw new Error("Error getting active application");
  }
  const data: AppResponse = await response.json();
  return data.active_app;
}

export async function getAvailableApps(): Promise<string[]> {
  const response = await fetch(`${API_URL}/apps/available`);
  if (!response.ok) {
    throw new Error("Error getting application list");
  }
  const data: AvailableAppsResponse = await response.json();
  return data.available_apps;
}

export async function activateApp(appName: string): Promise<string> {
  const response = await fetch(`${API_URL}/apps/activate/${appName}`, {
    method: "POST",
  });
  if (!response.ok) {
    throw new Error("Error activating application");
  }
  const data: ActivateAppResponse = await response.json();
  return data.active_app;
}
